npm/vite-dev-server/src/makeCypressPlugin.ts

```diff
@@ -48,40 +45,56 @@ export const makeCypressPlugin = (
 
   const posixSupportFilePath = supportFilePath ? convertPathToPosix(resolve(projectRoot, supportFilePath)) : undefined
 
-  const normalizedSupportFilePath = posixSupportFilePath ? `${base}@fs/${posixSupportFilePath}` : undefined
-
   return {
     name: pluginName,
     enforce: 'pre',
-    config (_, env) {
-      if (env) {
-        return {
-          define: {
-            'import.meta.env.__cypress_supportPath': JSON.stringify(normalizedSupportFilePath),
-            'import.meta.env.__cypress_originAutUrl': JSON.stringify(`__cypress/iframes/${convertPathToPosix(projectRoot)}/`),
-          },
-        }
-      }
-    },
     configResolved (config) {
       base = config.base
     },
-    transformIndexHtml () {
-      debug('transformIndexHtml with base', base)
-
-      return [
-        // load the script at the end of the body
-        // script has to be loaded when the vite client is connected
-        {
-          tag: 'script',
-          injectTo: 'body',
-          attrs: { type: 'module' },
-          children: `import(${JSON.stringify(`${base}@fs/${INIT_FILEPATH}`)})`,
-        },
-      ]
+    async transformIndexHtml () {
+      const indexHtmlPath = resolve(__dirname, '..', 'index.html')
+      const indexHtmlContent = await readFile(indexHtmlPath, { encoding: 'utf8' })
+      // find </body> last index
+      const endOfBody = indexHtmlContent.lastIndexOf('</body>')
+
+      // insert the script in the end of the body
+      return `${indexHtmlContent.substring(0, endOfBody)
+    }<script src="${base}cypress:client-init-test" type="module"></script>${
+      indexHtmlContent.substring(endOfBody)
+    }`
+    },
+    resolveId (id) {
+      if (id === 'cypress:config') {
+        return id
+      }
+
+      if (id === 'cypress:support-path') {
+        return posixSupportFilePath
+      }
+
+      if (id === 'cypress:spec-loaders') {
+        return id
+      }
+
+      if (id === '/cypress:client-init-test') {
+        return INIT_FILEPATH
+      }
+    },
+    load (id) {
+      if (id === 'cypress:spec-loaders') {
+        return `export default {\n${specs.map((s) => {
+          return `${JSON.stringify(s.relative)}:()=>import(${JSON.stringify(s.absolute)})`
+        }).join(',\n')}\n}`
+      }
+
+      if (id === 'cypress:config') {
+        return `
+export const supportPath = ${JSON.stringify(!!supportFilePath)}
```

Comment:
Won't this always eval to `export const supportPath = false`? Should it be

```js
${JSON.stringify(!!supportFilePath ? supportFilePath : false)}
```

Reply:
Agreed, what I meant was `hasSupportFile`